daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/unparsers/runtime1/BinaryNumberUnparsers.scala

```diff
@@ -163,7 +204,11 @@ class BinaryFloatUnparser(e: ElementRuntimeData) extends BinaryNumberBaseUnparse
     nBits: Int,
     finfo: FormatInfo
   ): Boolean = {
-    dos.putBinaryFloat(asFloat(value), finfo)
+    if (nBits > 0) {
```

Comment:
I believe binary float must be exactly 32 bits, anything else should be an error. In fact, I think we have a compile time restriction the binary float must have a known compile time length of 32 bits or its an SDE, so I think it's impossible for nBits to not be correct and we should just always call putBinaryFloat.